jenkins/run-server-func-tests

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+export EXTRA_PODMAN_SWITCHES=${EXTRA_PODMAN_SWITCHES:-"--pull=always -e=COV_REPORT_XML=yes"}
```

Comment:
Do we really want to be defining `COV_REPORT_XML=yes` in this context?  Are we really interested in code coverage for functional tests??  (And, how do we present this data without it messing up the unit test data?)

Reply:
Ha; code coverage is pretty much meaningless for functional tests since only the functional test client code is in scope. All the "interesting stuff" is in the remote containerized server.

Reply:
Huh.  I hadn't even considered that.  Yeah.

Although, having coverage data on the tests themselves isn't entirely useless.  😉

Reply:
See PR #3112.

Reply:
The question of whether we should enable coverage measurements for functional testing was not resolved in #3112.  (Or, rather, it was enabled there without further discussion....)

So..._does_ setting `EXTRA_PODMAN_SWITCHES` to include `-e=COV_REPORT_XML=yes` result in coverage data being generated?  Where is that data stored?  (E.g., is it stored in the transient file system in the container and then lost?)  It doesn't seem to adversely affect the unit test coverage data, but it also doesn't show up in it.

Is there any point to collecting it, or should we be disabling (or not enabling) it somewhere?  (Either here or in the caller of this script....)

Reply:
> So..._does_ setting `EXTRA_PODMAN_SWITCHES` to include `-e=COV_REPORT_XML=yes` result in coverage data being generated?

No, that value, `COV_REPORT_XML` is only consumed and used by `exec-unittests`.

> It doesn't seem to adversely affect the unit test coverage data, but it also doesn't show up in it.

Because the unit tests are executed using `exec-unittests` by `tox`.  These functional tests don't use that mechanism (see also PR #3123).

> Is there any point to collecting it, or should we be disabling (or not enabling) it somewhere? (Either here or in the caller of this script....)

There is no point to collecting it for functional tests at this point time, as this PR is not about to introduce a new mechanism for collecting it separate from the existing unit tests.  Perhaps something we can do in PR #3123 or in another PR separate from that.

I'll remove the reference.

Reply:
OK, good.  Now we just need to make the appropriate change to the Groovy file...but, as you say, perhaps we can do that in #3123 or elsewhere.

Reply:
It's needed at line 5 of the Groovy file because that what tells the unit tests executing in the Jenkins CI to emit an XML file instead of a HTML report directory hierarchy.  The XML file is what is consumed by the Cobertura / Jenkins integration to make the coverage table.  I don't think we can remove it from Groovy.

Reply:
Oh, no, we cannot remove it...but we might want to give it a better value just for the functional test invocation(s)....